File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    public double sideLen;

    public int getShape()
    {
        return 0;
    }

    public double[] getSize()
    {
        return new double[]{sideLen};
    }

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }
    
    public double getArea() {
        return sideLen * sideLen;
    }
}

File: typescript/src/shapes/square.ts
import { Shape,ShapeType } from "./shape.js";

function Square(sideLen: number): Shape {

    return {
        
        computeArea(): number {
            return sideLen*sideLen
        },
        getShape(): ShapeType {
            return 'circle'
          },
        getSize(): number[] {
            return [sideLen]
          },
        };
}



export { Square }

